tagger/interrogator.py

```diff
@@ -337,77 +434,129 @@ def large_batch_interrogate(self, images: List, dry_run=False) -> str:
+                  "lacks a repo_id. If not already local, download may fail.")
+
+        attrs = getattr(shared.opts, 'tagger_hf_hub_down_opts',
+                        f'cache_dir="{Its.hf_cache}"')
+        attrs = [attr.split('=') for attr in map(str.strip, attrs.split(','))]
+
+        signature = inspect.signature(hf_hub_download)
+        for arg, val in attrs:
+            if arg == 'filename' or arg in self.repo_specs:
+
+                print(f"Settings -> Tagger -> HuggingFace parameters: {arg}: "
+                      "Specific options need to go in the interrogators.json.")
+
+            elif arg in signature.parameters:
+                try:
+                    tp = signature.parameters[arg].annotation(val)
+                    self.hf_params[arg] = tp(val)
+
+                except TypeError:
+                    # unions, used for str or PathLike and a few.
+                    if val == 'None':
+                        self.hf_params[arg] = None
+                    elif arg == 'token' and val in {'True', 'False'}:
+                        self.hf_params[arg] = val == 'True'
+                    else:
+                        if val[0] == val[-1] and val[0] in "'\"":
+                            val = val[1:-1]
+                        self.hf_params[arg] = str(val)
             else:
-                package_name = "onnxruntime-gpu"
-            package = os.environ.get(
-                'ONNXRUNTIME_PACKAGE',
-                package_name
-            )
+                print(f"Settings -> Tagger -> HuggingFace parameters: {arg}: "
+                      "Invalid for hf_hub_download() => ignored.")
 
-            run_pip(f'install {package}', 'onnxruntime')
+    def download(self) -> Tuple[str, str]:
+        repo_id = self.hf_params.get('repo_id', '(?)')
+        print(f"Loading {self.name} model file from {repo_id}")
+
+        paths = [self.local_model, self.local_tags]
+
+        data = {}
+        for k in self.repo_specs:
+            if k in self.hf_params:
+                data[k] = self.hf_params[k]
+
+        # check if the model is up to date
+        info_path = Path(self.local_model).with_suffix('.info')
+        if info_path.exists():
+
+            if all(os.path.exists(p) for p in paths):
+                with open(info_path, 'r') as filen:
+                    try:
+                        old_data = json.load(filen)
+                        if old_data == data:
+                            print(f"Model {self.name} is up to date.")
+                            return paths
+                    except json.decoder.JSONDecodeError:
+                        pass
 
-    import onnxruntime
-    return onnxruntime
+            try:
+                for i, filen in enumerate([self.model_path, self.tags_path]):
+                    self.hf_params['filename'] = filen
+                    paths[i] = hf_hub_download(**self.hf_params)
+            except Exception as err:
```

Comment:
refer to [huggingface_hub/file_download.py](https://github.com/huggingface/huggingface_hub/blob/7981f7b2535238efdb68d3ef002c11ec9b67241b/src/huggingface_hub/file_download.py#L1243)
This error seems to have been caught.

It seems that it can be used offline without causing any errors, if the environment variable `HF_HUB_OFFLINE = True` is set, or `hf_hub_download(local_files_only = True)` is set.

Reply:
for the latter, this can now be set in Settings -> Tagger -> HuggingFace parameters (see other comment)